core/src/Cabana_CommunicationPlan.hpp

```diff
@@ -1083,7 +1084,8 @@ class CommunicationPlan
             for ( std::size_t j = 0; j < _num_export[i]; j++ )
             {
                 MPI_Irecv( export_indices.data() + idx, 1, MPI_INT,
-                           _neighbors[i], mpi_tag, comm(), &mpi_requests[idx] );
+                           _neighbors[i], mpi_tag + 1, comm(),
```

Comment:
Can you add a comment mentioning that incrementing the tag is a way to ensure the messages are processed in the right order? Probably just need one outside the loop in both functions